pkg/cloudprovider/providers/azure/azure_backoff.go

```diff
@@ -380,3 +393,36 @@ func isSuccessHTTPResponse(resp autorest.Response) bool {
 	}
 	return false
 }
+
+func shouldRetryHTTPRequest(resp *http.Response, err error) bool {
+	if err != nil {
+		return true
+	}
+
+	if resp != nil {
+		// HTTP 4xx or 5xx suggests we should retry
+		if 399 < resp.StatusCode && resp.StatusCode < 600 {
```

Comment:
If principal expired, we should surely retry the API call. Isn't this expected?

Reply:
if retry will exhaust API quota quicker, then we need to be more frugal.

Reply:
The logic here is same with shouldRetryAPIRequest, just with a different param (http.Response instead of autorest.Response). It doesn't change existing retry logic.